lib/common/zstd_internal.h

```diff
@@ -286,6 +286,28 @@ typedef struct {
     FSE_repeat litlength_repeatMode;
 } ZSTD_entropyCTables_t;
 
+struct ZSTD_CCtx_params_s {
+    ZSTD_compressionParameters cParams;
+    ZSTD_frameParameters fParams;
+
+    int compressionLevel;
+    U32 forceWindow;           /* force back-references to respect limit of
+                                * 1<<wLog, even for dictionary */
+    /* Dictionary */
+    U32 dictContentByRef;
```

Comment:
As mentioned, it feels strange to have this property "separated" from the pointer it's supposed to impact.
It feels better to keep it as a property of the `loadDictionary` command.

That means, either creating a new `ZSTD_CCtx_loadDictionary_byReference()`,
or changing `ZSTD_CCtx_loadDictionary()` prototype to add a loading parameter (`byCopy, byRef`),
or add a `ZSTD_CCtx_loadDictionary_advanced()` variant with said loading parameter.

For reference, our current `ZSTD_createCDict*()` API contains all 3 variants.

Reply:
If I understand correctly, we want to remove `ZSTD_p_refDictContent` completely, as well as `dictContentByRef` from both `ZSTD_CCtx_params` and `ZSTD_CCtx`?

Reply:
Yes